tests/integration/synapseclient/models/async/test_agent_async.py

```diff
@@ -0,0 +1,211 @@
+        test_prompt.session_id = test_session.id
+        # WHEN I send the job and wait for it to complete
+        await test_prompt.send_job_and_wait_async(
+            post_exchange_args={"newer_than": 0},
+            synapse_client=self.syn,
+        )
+        # THEN I expect the AgentPrompt to be updated with the response and trace
+        assert test_prompt.response is not None
+        assert test_prompt.trace is not None
+
+
+class TestAgentSession:
+    """Integration tests for the synchronous methods of the AgentSession class."""
+
+    @pytest.fixture(autouse=True, scope="function")
+    def init(self, syn: Synapse) -> None:
+        self.syn = syn
+
+    async def test_start(self) -> None:
+        # GIVEN an agent session with a valid agent registration id
+        agent_session = AgentSession(agent_registration_id=AGENT_REGISTRATION_ID)
+
+        # WHEN the start method is called
+        result_session = await agent_session.start_async(synapse_client=self.syn)
+
+        # THEN the result should be an AgentSession object
+        # with expected attributes including an empty chat history
+        assert result_session.id is not None
+        assert (
+            result_session.access_level == AgentSessionAccessLevel.PUBLICLY_ACCESSIBLE
+        )
+        assert result_session.started_on is not None
+        assert result_session.started_by is not None
+        assert result_session.modified_on is not None
+        assert result_session.agent_registration_id == AGENT_REGISTRATION_ID
+        assert result_session.etag is not None
+        assert result_session.chat_history == []
+
+    async def test_get(self) -> None:
+        # GIVEN an agent session with a valid agent registration id
+        agent_session = AgentSession(agent_registration_id=AGENT_REGISTRATION_ID)
+        # WHEN I start a session
+        await agent_session.start_async(synapse_client=self.syn)
+        # THEN I expect to be able to get the session with its id
+        new_session = await AgentSession(id=agent_session.id).get_async(
+            synapse_client=self.syn
+        )
+        assert new_session == agent_session
+
+    async def test_update(self) -> None:
+        # GIVEN an agent session with a valid agent
+        # registration id and access level set
+        agent_session = AgentSession(
+            agent_registration_id=AGENT_REGISTRATION_ID,
+            access_level=AgentSessionAccessLevel.PUBLICLY_ACCESSIBLE,
+        )
+        # WHEN I start a session
+        await agent_session.start_async(synapse_client=self.syn)
+        # AND I update the access level of the session
+        agent_session.access_level = AgentSessionAccessLevel.READ_YOUR_PRIVATE_DATA
+        await agent_session.update_async(synapse_client=self.syn)
+        # THEN I expect the access level to be updated
+        assert (
+            agent_session.access_level == AgentSessionAccessLevel.READ_YOUR_PRIVATE_DATA
+        )
```

Comment:
A better verification of this is to make sure that the data was stored in Synapse. Consider if this silently failed and did not persist the data to Synapse. Your instance would have the updated attribute value, but the next time the session is retrieved it could be using the previous access level.

Perform a new read with a new instance after calling `update_async` and verify on that newly read instance that it contains the right access_level